packages/components/organisms/f-checkout/src/components/Selector.vue

```diff
@@ -40,36 +40,48 @@ export default {
         * Returns an array of formatted dates to display as options in a dropdown
         */
         fulfilmentTimes () {
-            const times = this.availableFulfilment.isAsapAvailable
-                ? [this.$t('asapFulfilmentOption')]
-                : [];
-
+            const times = [];
             this.availableFulfilment.times.forEach(time => {
                 const formattedTime = this.$d(new Date(time.from), 'short');
 
-                times.push(formattedTime);
+                times.push({
+                    text: formattedTime,
+                    value: time.from
+                });
             });
 
+            if (this.availableFulfilment.isAsapAvailable && times.length) {
+                times[0].text = this.$t('asapFulfilmentOption');
+            }
+
             return times;
         }
     },
 
     watch: {
         'fulfilmentTimes' () {
             if (this.fulfilmentTimes.length > 0) {
```

Comment:
Doesn't have to change here but can just be `this.fulfilmentTimes.length` as the length check will return a boolean.